libraries/src/Exception/ExceptionHandler.php

```
@@ -163,30 +163,39 @@ public static function render(\Throwable $error)
 
 		/*
 		 * To reach this point in the code means there was an error creating the error page.
-		 * We try to send at least something back other than a WSOD at this point.
+		 *
+		 * Let global handler to handle the error, @see bootstrap.php
 		 */
-		if (!$isCli && !headers_sent())
+		if (isset($e))
 		{
-			header('HTTP/1.1 500 Internal Server Error');
-		}
-
-		$message = 'Error';
-
-		// Make sure we do not display sensitive data in production environments
-		if (ini_get('display_errors'))
-		{
-			$message .= ': ';
-
-			if (isset($e))
+			/*
+			 * Here the thing, at this point we have 2 exceptions:
+			 * $e - the error caused by error renderer
+			 * $error - the main error
+			 *
+			 * Both we need to show without loosing of a trace information
+			 * So use a bit of magic to merge them
+			 */
+			try
 			{
-				$message .= $e->getMessage() . ': ';
+				try
+				{
+					throw $error;
+				}
+				finally
+				{
+					throw $e;
```

You’re better off using Reflection to alter the Exception properties than trying this black hat trickery.  No way anyone can actually read and understand what this try/finally block is actually doing.

---

Reflection was a first thing I tried, but I got another Exception "Property previous not exist" when doing:
```
$refl = new ReflectionClass($e);
$property = $refl->getProperty('previous');
```
Try/finally is one that work, it "collect"  exceptions

---

That’s not a valid way of “collecting” Exceptions.  At that point one can not evaluate the code and determine what is the intended order of operations or the intended end result, you’re basically relying on some magic and probably undocumented PHP behavior and that is too risky.

You’re better off rethrowing the original Exception without the error page exception than trying this trickery.

---

You just not like a magic :-p
Because it not documented it does not means that it "not a valid", there many stuf undocumented, whole "finally" documentation has only 1-2 sentence.
As long as it works and allow to see nested exceptions it is okay, this is better than hiding errors that not possible to debug later.
I do not see any big risk in this particular case.

---

Relying on magic behaviors or undocumented engine behaviors is what gets you in trouble later on.  Code is ALWAYS more stable if it has a predictable behavior, AFAIK this behavior is not specified in the RFC introducing finally support to PHP or any documentation detailing its use.  Inherently this means it is a risky behavior to use as it may very well not be an engine design but something that works by accident in the way you want it to.

---

I found at least 2 test cases for this behavior:
exactly for this case : https://github.com/php/php-src/blob/master/Zend/tests/try/try_finally_002.phpt
and more complex case: https://github.com/php/php-src/blob/master/Zend/tests/try/catch_finally_004.phpt

Does not looks like a random engine behavior, which has test cases, or? 
This part of exception nesting API. There zero detail about it, but it does not means that it wrong.
I think it pretty stable code to use, I will leave it.
One issue I see that not all may understood how it work, but that not a code problem.

---

If you’re going to insist on keeping it this way (I’m not a fan of it but whatever) then the comment block for this needs to be a lot more clear about what it’s doing then.  Because it’s not really helping to explain what you’re doing here at all or what the end result you’re creating is.